src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs

```diff
@@ -213,5 +237,60 @@ public void RequestTreeNavigateTo(IControl control, bool isVisualTree)
                 tree.SelectControl(control);
             }
         }
+
+        [DependsOn(nameof(TreePageViewModel.SelectedNode))]
+        [DependsOn(nameof(Content))]
+        bool CanShot(object? paramter)
+        {
+            return Content is TreePageViewModel tree
+                && tree.SelectedNode != null
+                && tree.SelectedNode.Visual != null
+                && tree.SelectedNode.Visual.VisualRoot != null;
+        }
+
+        void Shot(object? parameter)
+        {
+            // This is a workaround because MethodToCommand does not support the asynchronous method.
```

Comment:
Doesn't "async void" works? Or just a command with async void handler.

Reply:
fixed

Reply:
My initial question was about that entire method. Why we need Task.Run and Dispatcher.UIThread.Post inside?
As far as I expect, output.Dispose(); call will be the slowest, because it needs to flush content. And there should be FlushAsync or something, so we can avoid Task.Run (and dispatcher) at all there.

Reply:
Sorry I did not understand.
I made this choice because CreateDirectory and new System.IO.FileStream can also take time.
While Dispatcher.UIThread.Post to make sure it runs in the UIThread.